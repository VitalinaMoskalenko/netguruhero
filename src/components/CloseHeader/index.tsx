import React from "react";
import styled from "styled-components";
import { Body } from "../Paragraphs";
import { H3 } from "../Headings";

type PropsType = {
  title?: string;
  onClose: () => void;
};

type ContainerPropsType = {
  hasTitle: boolean;
};

const Container = styled.div<ContainerPropsType>`
  display: flex;
  justify-content: ${({ hasTitle }) =>
    hasTitle ? "space-between" : "flex-end"};
`;
const CloseButton = styled.button`
  cursor: pointer;
  border: 0px;
  background-color: ${({ theme }) => theme.colors.gray};
`;
const Close = styled(Body)`
  color: ${({ theme }) => theme.colors.black05};
`;

const CloseHeader = ({ title, onClose }: PropsType) => {
  return (
    <Container hasTitle={!!title}>
      {title && <H3>{title}</H3>}
      <CloseButton onClick={onClose}>
        <Close>x</Close>
      </CloseButton>
    </Container>
  );
};

export default CloseHeader;
