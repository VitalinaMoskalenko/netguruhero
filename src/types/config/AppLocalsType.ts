export enum AppLocalsType {
  en = "en",
}
