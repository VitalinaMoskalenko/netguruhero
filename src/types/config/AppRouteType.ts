export enum AppRouteType {
  home = "/",
  addHeroForm = "/addHeroForm",
  heroDetails = "/heroDetails",
  heroDetailsWithId = "/heroDetails/:id",
}
