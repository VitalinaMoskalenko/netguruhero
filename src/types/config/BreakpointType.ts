export enum BreakpointType {
  xxs = 375,
  xs = 768,
  xxl = 1440,
}
