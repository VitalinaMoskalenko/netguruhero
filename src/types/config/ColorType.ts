export enum ColorType {
  white = "white",
  black = "black",
  green = "#63cd93",
  blue = "#6199f3",
  gray = "#f5f6fb",
  black05 = "rgba(0,0,0, 0.5)",
}
