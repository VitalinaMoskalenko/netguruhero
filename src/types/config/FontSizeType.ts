export enum FontSizeType {
  smallBody = 14,
  body = 18,
  h5 = 16,
  h4 = 18,
  h3 = 24,
  h2 = 32,
  h1 = 48,
}
