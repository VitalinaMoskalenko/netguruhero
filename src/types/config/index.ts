export * from "./AppRouteType";
export * from "./FontSizeType";
export * from "./ColorType";
export * from "./BreakpointType";
export * from "./AppLocalsType";
