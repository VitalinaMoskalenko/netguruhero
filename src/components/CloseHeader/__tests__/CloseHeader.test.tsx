import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import CloseHeader from "../";
import { theme } from "../../../config";

it("renders close header correctly", () => {
  const closeHeaderComponent = renderer
    .create(
      <ThemeProvider theme={theme}>
        <CloseHeader title="CloseHeader" onClose={() => {}} />
      </ThemeProvider>
    )
    .toJSON();

  expect(closeHeaderComponent).toMatchSnapshot();
});
