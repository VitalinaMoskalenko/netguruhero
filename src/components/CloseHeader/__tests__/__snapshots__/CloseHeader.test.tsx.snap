// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders close header correctly 1`] = `
<div
  className="sc-iCoGMd bpOkbK"
>
  <h1
    className="sc-eCApnc TwDaM"
  >
    CloseHeader
  </h1>
  <button
    className="sc-fujyAs BQeBv"
    onClick={[Function]}
  >
    <p
      className="sc-bdnxRM sc-pNWdM jupkGm lpcoCX"
    >
      x
    </p>
  </button>
</div>
`;
